### ISSUE TITLE
bug: Schema integrity check fails when you have 2 schema nodes defined with the same name but different namespace

### ISSUE THREAD

**Issue by wvandeun** [opsmill/infrahub#3986]
### Component

API Server / GraphQL

### Infrahub version

0.16.0

### Current Behavior

When you define 2 nodes in the schema with the same name, but a different namespace then the schema integrity check fails during the proposed change.

![image](https://github.com/user-attachments/assets/194acfeb-a6d1-4019-853b-9095629f15ab)

`infrahubctl schema load` or `infrahubctl schema check` do not complain about the schema.

### Expected Behavior

The schema integrity check should succeed since the nodes have been defined in a different namespace.

### Steps to Reproduce

- create a branch
- load this schema
```yaml
---
version: "1.0"
nodes:
  - name: Device
    namespace: Zorro
    label: "Device"
    display_labels:
      - name__value
    attributes:
      - name: name
        kind: Text
        unique: true
      - name: description
        kind: Text
        optional: true
  - name: Device
    namespace: Infra
    label: "Device"
    display_labels:
      - name__value
    attributes:
      - name: name
        kind: Text
        unique: true
      - name: description
        kind: Text
        optional: true
```
- create a proposed change to merge the branch into main
- schema integrity check will fail

**Comment by wvandeun** [comment]
The same issue occurs when you have 2 generics defined in your schema that have the same name, but use a different namespace.

**Comment by ogenstad** [comment]
Fixed in #4029.
